lib/internal/streams/destroy.js

```diff
@@ -62,10 +62,10 @@ function _destroy(self, err, cb) {
 
     if (err) {
       if (w) {
-        w.errored = true;
+        w.errored = err;
       }
       if (r) {
-        r.errored = true;
+        r.errored = err;
       }
     }
```

Comment:
@addaleax I think I need a little help. Keeping a reference to the error on the stream object seem to cause `JSStreamSocket` (or at least the `TLSSocket`) to not get GC:d. Is this triggering some other GC liveness bug?

```js
AssertionError [ERR_ASSERTION]: 0 < 256
    at TLSSocket.connect (/Users/ronagy/GitHub/nxtedition/node/test/parallel/test-gc-tls-external-memory.js:36:5)
    at TLSSocket.<anonymous> (/Users/ronagy/GitHub/nxtedition/node/test/common/index.js:365:15)
    at TLSSocket.emit (events.js:314:20)
    at emitErrorNT (internal/streams/destroy.js:127:8)
    at emitErrorCloseNT (internal/streams/destroy.js:92:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at runNextTicks (internal/process/task_queues.js:62:3)
    at processImmediate (internal/timers.js:435:9) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '=='
}
```

Reply:
Moved to https://github.com/nodejs/node/pull/34103